include/chimera/util.h

```diff
@@ -46,6 +46,35 @@ void extendWithYAMLNode(::mstch::map &map, const YAML::Node &node,
                         bool overwrite = false,
                         ScalarConversionFn fn = nullptr);
 
+/**
+ * Returns a YAML node for a key in @c node. Returns an invalid node if @c node
+ * is invalid or @c key is doesn't exist in @node.
+ */
+YAML::Node lookupYAMLNode(const YAML::Node &node, const std::string &key);
+
+/**
+ * Returns a nested YAML node for a series of keys. Returns an invalid nod if
+ * @c node is invalid or failed to find any nested YAML node for a key in @c
+ * keys.
+ */
+template <typename... Args>
+YAML::Node lookupYAMLNode(const YAML::Node &node, const std::string &key,
+                          Args &&... args)
+{
+    // Return if 'node' is invalid
+    if (not node)
+        return node;
+
+    auto next = node[key];
+
+    // Return if failed to find a tag of 'key'
+    if (not next)
+        return next;
+
+    // Lookup for the next nested tags
+    return lookupYAMLNode(next, std::forward<Args>(args)...);
+}
```

Comment:
@psigen If you don't prefer in-line function definitions, then we could consider moving the definition to the bottom of this header (keeping the declaration here) or to a new header (e.g., `util_impl.h`)

Reply:
I'm ok with this as is, there isn't much in this file.